pyproj/geod.py

```diff
@@ -298,6 +298,15 @@ def fwd(  # pylint: disable=invalid-name
         scalar or array:
             Back azimuth(s)
         """
+        if not inplace:
```

Comment:
I don't think this `inplace` check is needed. The reason is that inplace still won't work for scalars in the latter code.

Reply:
I thought the same thing, but the test for `inplace` types fails with numpy arrays of length-1.

Maybe Cython does some auto-casting and allows `np.array([1], dtype=np.float64)` to pass into a `double x` function?

Reply:
That is a gotcha, but kind of makes sense. This is concerning for any array of length 1 as it will be passed into this function by default and get back a scalar instead of an array. Any thoughts on how to handle that?

Reply:
Hm... I'm not sure how to prevent the cast/raise and a quick search didn't turn up anything super obvious.

I think this is only an issue for numpy-like arrays?
```python
float(np.array([0]))  # scalar
float(array.array("d", [0]))  # ValueError
```

Reply:
That is really unfortunate that it is the case. I wonder why numpy allows that.
Due to this behavior, I am thinking it would be better to have a separate interface for the optimized point functions.

Reply:
Yes, this is unfortunate :( Apparently you can typecheck during a typecast in cython `<MyType?>my_type`, but I can't figure out how to do that via the function call... I am not knowledgeable in Cython at all.

Another option which adds a small bit of overhead, but can get rid of the try/except is:
```python
        if (isinstance(lons, (float, int))
            and isinstance(lats, (float, int))
            and isinstance(az, (float, int))
            and isinstance(dist, (float, int))):
```

what are your thoughts about that as an option versus the extra function? (~800 ns in this implementation versus 600 ns with the try/except)

`all(...)` adds a bit of overhead as well, so it is better to check each arg explicitly.

Reply:
https://cython.readthedocs.io/en/latest/src/userguide/extension_types.html#type-testing-and-casting

> To explicitly test the type of an object, use the isinstance() builtin function. For known builtin or extension types, Cython translates these into a fast and safe type check that ignores changes to the object’s __class__ attribute etc., so that after a successful isinstance() test, code can rely on the expected C structure of the extension type and its C-level attributes (stored in the object’s C struct) and [cdef](https://cython.readthedocs.io/en/latest/src/userguide/language_basics.html#cdef)/@cfunc methods.

Reply:
> what are your thoughts about that as an option versus the extra function? (~800 ns in this implementation versus 600 ns with the try/except)

Looks like checking for `float`|`int` types may not be desirable if you are coming from numpy:

```python
>>> import numbers
>>> isinstance(2, numbers.Real)
True
>>> import numpy
>>> isinstance(numpy.int64(2), numbers.Real)
True
>>> isinstance(numpy.array([2]), numbers.Real)
False
>>> isinstance(numpy.int64(2), int)
False
```

Reply:
Inside Cython, if I declare the inputs to the function as `double lons` and then `print(type(lons))` it is always class float, so it has already been cast upon entering the function. Is that suggesting to use `object lons` and then `if not isinstance(lons, (float, int)): raise TypeError; else: double lon = <double>lons`? That seems pretty verbose, but doable if we want.

I'm not following why that the typecheck from numpy would be undesirable? Worst case if you're passing in a numpy scalar int, then you'll just go down the old path because you've failed the isinstance check. If you pass in a `np.float64(2)` numpy scalar, we will return a standard Python scalar instead, but it looks like that already happens in the current methods.

I just pushed up the isinstance() checks from the Python side in a separate commit.

Reply:
>  Is that suggesting to use object lons and then if not isinstance(lons, (float, int)): raise TypeError; else: double lon = <double>lons? That seems pretty verbose, but doable if we want.

Does that have any significant impact on performance?

Reply:
I think it is slightly faster to do that, back down around 600ish ns again. Although the code isn't quite as nice in the C layer, but that is probably OK.